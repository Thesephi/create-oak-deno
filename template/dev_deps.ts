export { superoak } from "https://deno.land/x/superoak@4.8.1/mod.ts";
export { assertSnapshot } from "jsr:@std/testing@1.0.5/snapshot";
