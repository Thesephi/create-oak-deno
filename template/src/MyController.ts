import { Controller, Get } from "@dklab/oak-routing-ctrl";
import type { Context, RouteContext } from "@oak/oak";

@Controller("/v1")
export class MyController {
  @Get("/hello/:name")
  hello(ctx: Context & RouteContext<"/hello/:name">) {
    return `hello, ${ctx.params.name}`;
  }
}
