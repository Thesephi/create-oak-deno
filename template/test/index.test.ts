import { assertSnapshot, superoak } from "../dev_deps.ts";
import { Application } from "@oak/oak";
import { useOakServer, useOas } from "@dklab/oak-routing-ctrl";
import { MyController } from "../src/MyController.ts";

const app = new Application();
useOakServer(app, [MyController]);
useOas(app);

Deno.test("call /v1/hello/:name", async () => {
  const request = await superoak(app);
  await request.get("/v1/hello/tester").expect("hello, tester");
});

Deno.test("call /oas.json", async (t) => {
  const request = await superoak(app);
  const resp = await request.get("/oas.json");
  const respBody = await resp.body;
  await assertSnapshot(t, respBody);
});
