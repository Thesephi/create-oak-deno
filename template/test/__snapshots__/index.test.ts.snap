export const snapshot = {};

snapshot[`call /oas.json 1`] = `
{
  components: {
    parameters: {},
    schemas: {},
  },
  info: {
    description: "This is the API",
    title: "My API",
    version: "1.0.0",
  },
  openapi: "3.0.0",
  paths: {
    "/v1/hello/{name}": {
      get: {
        parameters: [
          {
            in: "path",
            name: "name",
            required: true,
            schema: {
              type: "string",
            },
          },
        ],
        responses: {
          "200": {
            content: {
              "text/html": {
                schema: {
                  type: "string",
                },
              },
            },
            description: "Success",
          },
        },
      },
    },
  },
  servers: [
    {
      url: "/",
    },
  ],
}
`;
